server/pkg/pipeline/queue.go

```diff
@@ -78,6 +80,12 @@ func (q Queue[T]) Enqueue(ctx context.Context, item T) {
 		return
 	}
 
+	workerCtx := context.Background()
+	if propagator := otel.GetTextMapPropagator(); propagator != nil {
+		var carrier propagation.HeaderCarrier
+		workerCtx = propagator.Extract(workerCtx, carrier)
+	}
```

Comment:
I need a new context here because `ctx` will be canceled right after this function returns.